packages/ember-glimmer/tests/integration/syntax/with-test.js

```diff
@@ -30,4 +38,246 @@ moduleFor('Syntax test: {{#with}}', class extends SharedConditionalsTest {
+      set(this.context, 'user', { name: 'Yehuda Katz' });
+    });
+
+    this.assertText('Admin: Tom Dale User: Yehuda Katz');
+  }
+
+  ['@test the scoped variable is not available outside the {{with}} block.']() {
+    this.render(`{{#with first as |ring|}}{{ring}}-{{#with fifth as |ring|}}{{ring}}-{{#with ninth as |ring|}}{{ring}}-{{/with}}{{ring}}-{{/with}}{{ring}}{{/with}}`, {
+      first: 'Limbo',
+      fifth: 'Wrath',
+      ninth: 'Treachery'
+    });
+
+    this.assertText('Limbo-Wrath-Treachery-Wrath-Limbo');
+
+    this.runTask(() => this.rerender());
+
+    this.assertText('Limbo-Wrath-Treachery-Wrath-Limbo');
+
+    this.runTask(() => {
+      set(this.context, 'first', 'I');
+      set(this.context, 'fifth', 'D');
+      set(this.context, 'ninth', 'K');
+    });
+
+    this.assertText('I-D-K-D-I');
+
+    this.runTask(() => {
+      set(this.context, 'first', 'Limbo');
+      set(this.context, 'fifth', 'Wrath');
+      set(this.context, 'ninth', 'Treachery');
+    });
+
+    this.assertText('Limbo-Wrath-Treachery-Wrath-Limbo');
+  }
+
+  ['@test it should support #with name as |foo|, then #with foo as |bar|']() {
+    this.render(`{{#with name as |foo|}}{{#with foo as |bar|}}{{bar}}{{/with}}{{/with}}`, {
+      name: 'caterpillar'
+    });
+
+    this.assertText('caterpillar');
+
+    this.runTask(() => this.rerender());
+
+    this.assertText('caterpillar');
+
+    this.runTask(() => {
+      set(this.context, 'name', 'butterfly');
+    });
+
+    this.assertText('butterfly');
+
+    this.runTask(() => set(this.context, 'name', 'caterpillar'));
+
+    this.assertText('caterpillar');
+  }
+
+  ['@test nested {{with}} blocks shadow the outer scoped variable properly.']() {
+    this.render(`{{#with first as |ring|}}{{ring}}-{{#with fifth as |ring|}}{{ring}}-{{#with ninth as |ring|}}{{ring}}-{{/with}}{{ring}}-{{/with}}{{ring}}{{/with}}`, {
+      first: 'Limbo',
+      fifth: 'Wrath',
+      ninth: 'Treachery'
+    });
+
+    this.assertText('Limbo-Wrath-Treachery-Wrath-Limbo');
+
+    this.runTask(() => this.rerender());
+
+    this.runTask(() => {
+      set(this.context, 'first', 'I');
+      set(this.context, 'fifth', 'D');
+      set(this.context, 'ninth', 'K');
+    });
+
+    this.assertText('I-D-K-D-I');
+
+    this.runTask(() => {
+      set(this.context, 'first', 'Limbo');
+      set(this.context, 'fifth', 'Wrath');
+      set(this.context, 'ninth', 'Treachery');
+    });
+
+    this.assertText('Limbo-Wrath-Treachery-Wrath-Limbo');
+  }
+});
+
+moduleFor('Syntax test: {{#with this as |foo|}} ', class extends RenderingTest {
+
+  ['@test it should support #with this as |qux|']() {
+    this.render(`{{#with this as |person|}}{{person.name}}{{/with}}`, {
+      name: 'Los Pivots'
+    });
+
+    this.assertText('Los Pivots');
+
+    this.runTask(() => this.rerender());
+
+    this.assertText('Los Pivots');
+
+    this.runTask(() => {
+      set(this.context, 'name', 'l\'Pivots');
+    });
```

Comment:
[nit] could be on a single line (also, could use backticks to avoid the escaping, but I'm not sure if that's what we usually do for ember)